src/backend/catalog/aclchk.c

```diff
@@ -224,16 +224,17 @@ restrict_and_check_grant(bool is_grant, AclMode avail_goptions, bool all_privs,
 	 * If we found no grant options, consider whether to issue a hard error.
 	 * Per spec, having any privilege at all on the object will get you by
 	 * here.
+	 * Superusers or on QE bypass all permission checking.
 	 */
-	if (avail_goptions == ACL_NO_RIGHTS)
+	if (avail_goptions == ACL_NO_RIGHTS && GP_ROLE_EXECUTE != Gp_role)
```

Comment:
I think you  should change to "Gp_role != GP_ROLE_EXECUTE"
This is more human-understandable and follows the code convention.